code/game/objects/items/storage/holsters.dm

```diff
@@ -0,0 +1,254 @@
+///Parent item for all holster type storage items
+
+/obj/item/storage/holster
+	name = "holster"
+	desc = "Holds stuff, and sometimes goes swoosh."
+	icon_state = "backpack"
+	w_class = WEIGHT_CLASS_BULKY
+	max_w_class = 4 ///normally the special item will be larger than what should fit. Child items will have lower limits and an override
+	storage_slots = 1
+	max_storage_space = 4
+	flags_equip_slot = ITEM_SLOT_BACK
+	draw_mode = 1
+	allow_drawing_method = TRUE
+	///is used to store the 'empty' sprite name
+	var/base_icon = "m37_holster"
+	///the sound produced when the special item is drawn
+	var/drawSound = 'sound/weapons/guns/misc/rifle_draw.ogg'
+	///the sound produced when the special item is sheathed
+	var/sheatheSound = 'sound/weapons/guns/misc/rifle_draw.ogg'
+	///the snowflake item(s) that will update the sprite.
+	var/list/holsterable_allowed = list()
+	///records whether the special item currently in the holster
+	var/holstered = FALSE
+
+/obj/item/storage/holster/equipped(mob/user, slot)
+	if(slot == SLOT_BACK || slot == SLOT_BELT || slot == SLOT_S_STORE)	///add more if needed
+		mouse_opacity = 2 //so it's easier to click when properly equipped.
+	..()
+
+/obj/item/storage/holster/dropped(mob/user)
+	mouse_opacity = initial(mouse_opacity)
+	..()
+
+/obj/item/storage/holster/should_access_delay(obj/item/item, mob/user, taking_out) ///defaults to 0
+	if(!taking_out) // Always allow items to be tossed in instantly
+		return FALSE
+	if(ishuman(user))
+		var/mob/living/carbon/human/human_user = user
+		if(human_user.back == src)
+			return TRUE
+	return FALSE
+
+///check to see if the item being inserted is the snowflake item
```

Comment:
dont need to autodoc an already defined proc.
This should be a comment.

Reply:
done